tests/all/memory.rs

```diff
@@ -375,7 +375,7 @@ fn massive_64_bit_still_limited() -> Result<()> {
             _max: Option<usize>,
         ) -> Result<bool> {
             self.hit = true;
-            Ok(true)
+            Ok(false)
```

Comment:
I think this'll want to stay as `true` as it's trying to exercise the path where the OS says "no you can't have that much memory" as opposed to the store limiting memory, and if `false` is returned here we won't ever hit the OS. I'm not sure why this test previously passed on MIRI though and it's failing on this PR

Reply:
> I'm not sure why this test previously passed on MIRI though and it's failing on this PR

I'm not sure either, AFAICT we should be doing the same thing as before.